rpcs3/Emu/System.h

```diff
@@ -301,8 +301,8 @@ class Emulator final
 	bool Pause(bool freeze_emulation = false);
 	void Resume();
 	void GracefulShutdown(bool allow_autoexit = true, bool async_op = false, bool savestate = false);
-	void Kill(bool allow_autoexit = true, bool savestate = false);
-	game_boot_result Restart(bool savestate = false);
+	std::shared_ptr<utils::serial> Kill(bool allow_autoexit = true, bool savestate = false);
```

Comment:
There doesn't seem to be a reason to return a pointer here.
It's also really confusing as always

Reply:
This is for future functionality.

Reply:
The reason that m_ar is no longer used for it is because it can leak without management, here if it's unused it deallocates itself.